validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorStatusLogger.java

```diff
@@ -0,0 +1,135 @@
+              if (maybeValidatorStatuses.isEmpty()) {
+                LOG.error("Unable to retrieve validator statuses from BeaconNode.");
+                return;
+              }
+
+              Map<BLSPublicKey, ValidatorStatus> validatorStatuses = maybeValidatorStatuses.get();
+              if (validatorPublicKeys.size() < VALIDATOR_KEYS_PRINT_LIMIT) {
+                printValidatorStatusesOneByOne(validatorStatuses);
+              } else {
+                printValidatorStatusSummary(validatorStatuses);
+              }
+            })
+        .reportExceptions();
+  }
+
+  private void printValidatorStatusesOneByOne(
+      Map<BLSPublicKey, ValidatorStatus> validatorStatuses) {
+    for (BLSPublicKey publicKey : validatorPublicKeys) {
+      Optional<ValidatorStatus> maybeValidatorStatus =
+          Optional.ofNullable(validatorStatuses.get(publicKey));
+      maybeValidatorStatus.ifPresentOrElse(
+          validatorStatus ->
+              LOG.info(
+                  "Validator {} status is " + validatorStatus, publicKey.toAbbreviatedString()),
+          () -> LOG.error("Error retrieving status for validator {}", publicKey));
+    }
+  }
+
+  private void printValidatorStatusSummary(Map<BLSPublicKey, ValidatorStatus> validatorStatuses) {
+    Map<ValidatorStatus, AtomicInteger> validatorStatusCount = new HashMap<>();
+    for (ValidatorStatus status : validatorStatuses.values()) {
+      AtomicInteger count =
+          validatorStatusCount.computeIfAbsent(status, __ -> new AtomicInteger(0));
+      count.incrementAndGet();
+    }
+
+    for (Map.Entry<ValidatorStatus, AtomicInteger> statusCount : validatorStatusCount.entrySet()) {
+      LOG.info(
+          statusCount.getValue().get()
+              + " validators are in "
+              + statusCount.getKey().name()
+              + " state.");
+    }
+  }
+
+  public void checkValidatorStatusChanges() {
+    validatorApiChannel
+        .getValidatorStatuses(getAsIdentifiers(validatorPublicKeys))
+        .thenAccept(
+            maybeNewValidatorStatuses -> {
+              if (maybeNewValidatorStatuses.isEmpty()) {
+                LOG.error("Unable to retrieve validator statuses from BeaconNode.");
+                return;
+              }
+
+              Map<BLSPublicKey, ValidatorStatus> newValidatorStatuses =
+                  maybeNewValidatorStatuses.get();
+
+              Map<BLSPublicKey, ValidatorStatus> oldValidatorStatuses =
+                  latestValidatorStatuses.getAndSet(newValidatorStatuses);
+              if (oldValidatorStatuses == null) {
+                return;
+              }
+
+              for (BLSPublicKey key : oldValidatorStatuses.keySet()) {
+                ValidatorStatus oldStatus = oldValidatorStatuses.get(key);
+                ValidatorStatus newStatus = newValidatorStatuses.get(key);
+                if (oldStatus.equals(newStatus)) {
+                  continue;
+                }
+
+                LOG.warn(
```

Comment:
Should this be at INFO level unless the change is concerning (if the validator was slashed, for example)?

Reply:
I think a validator status change is important enough to merit a log.warn regardless, since there is actual money on the line. 

The validator might have changed (or might be seen from Teku's perspective to have changed) in an error-prone way, and this could help operators see and fix that.